raiden/ui/prompt.py

```diff
@@ -1,89 +1,85 @@
-        if address_hex is not None:
-            click.secho(
-                f"Account '{address_hex}' could not be found on the system. Aborting ...",
-                fg='red',
-            )
-            sys.exit(1)
 
-        addresses = list(accmgr.accounts.keys())
-        formatted_addresses = [
-            '[{:3d}] - {}'.format(idx, to_checksum_address(addr))
-            for idx, addr in enumerate(addresses)
-        ]
+def prompt_account(account_manager: AccountManager) -> AddressHex:
+    addresses = list(account_manager.accounts.keys())
+    formatted_addresses = [
+        '[{:3d}] - {}'.format(idx, to_checksum_address(addr))
+        for idx, addr in enumerate(addresses)
+    ]
+
+    print('The following accounts were found in your machine:')
+    print('')
+    print('\n'.join(formatted_addresses))
+    print('')
 
-        should_prompt = True
+    while True:
+        idx = click.prompt('Select one of them by index to continue', type=int)
 
-        print('The following accounts were found in your machine:')
-        print('')
-        print('\n'.join(formatted_addresses))
-        print('')
+        if 0 <= idx < len(addresses):
+            return addresses[idx]
 
-        while should_prompt:
-            idx = click.prompt('Select one of them by index to continue', type=int)
+        print('\nError: Provided index "{}" is out of bounds\n'.format(idx))
 
-            if 0 <= idx < len(addresses):
-                should_prompt = False
-            else:
-                print('\nError: Provided index "{}" is out of bounds\n'.format(idx))
 
-        address_hex = addresses[idx]
+def unlock_account_with_passwordfile(
+        account_manager: AccountManager,
+        address_hex: AddressHex,
+        password_file: io.RawIOBase,
+) -> PrivateKey:
+    password = password_file.read()
 
-    password = None
-    if password_file:
-        password = password_file.read()
-        if password != '':
-            password = password.splitlines()[0]
-    if password is not None:
+    try:
+        return account_manager.get_privkey(address_hex, password)
+    except ValueError:
+        click.secho(
+            f'Incorrect password for {address_hex} in file. Aborting ...',
+            fg='red',
+        )
+        sys.exit(1)
+
+
+def unlock_account_with_passwordprompt(
+        account_manager: AccountManager,
+        address_hex: AddressHex,
+) -> PrivateKey:
+    while True:
         try:
-            privatekey_bin = accmgr.get_privkey(address_hex, password)
+            password = getpass.getpass(
+                f'Enter the password to unlock {address_hex}: ',
+            )
+            return account_manager.get_privkey(address_hex, password)
         except ValueError:
-            # ValueError exception raised if the password is incorrect
-            click.secho(
-                f'Incorrect password for {address_hex} in file. Aborting ...',
-                fg='red',
+            print(
```

Comment:
There's no breaking out of the loop here... this could go on forever

Reply:
Oh, I assumed the message was good enough, I didn't think about the user being stuck and not knowing how to quit. Should I revert it?

Reply:
done